informative-indexer/flusher/processors/validator/processor.go

```diff
@@ -21,13 +23,18 @@ var _ processors.Processor = &Processor{}
 func (p *Processor) InitProcessor() {
 	p.stakeChanges = make(map[string]int64)
 	p.validators = make(map[string]bool)
+	p.slashEvents = make([]db.ValidatorSlashEvent, 0)
 }
 
 func (p *Processor) Name() string {
 	return "validator"
 }
 
-func (p *Processor) ProcessTxEvents(txResult *mq.TxResult, height int64, stateUpdateManager *statetracker.StateUpdateManager, _ *db.Transaction) error {
+func (p *Processor) ProcessTransaction(txResult *mq.TxResult, height int64, stateUpdateManager *statetracker.StateUpdateManager, _ *db.Transaction) error {
```

Comment:
could we abstract three steps into the interface so the caller will call the same for every processors